examples/chip-tool/commands/tests/TestCommand.cpp

```diff
@@ -45,3 +47,17 @@ void TestCommand::OnDeviceConnectionFailureFn(void * context, NodeId deviceId, C
     VerifyOrReturn(command != nullptr, ChipLogError(chipTool, "Test command context is null"));
     command->SetCommandExitStatus(error);
 }
+
+CHIP_ERROR TestCommand::WaitForMs(unsigned int ms)
+{
+    void * context = this;
+    std::thread waitingThread([context, ms] {
+        std::this_thread::sleep_for(std::chrono::milliseconds(ms));
+        auto * command = static_cast<TestCommand *>(context);
+        command->NextTest();
```

Comment:
This ends up running `command->NextTest()` on that helper thread, right?  That's presumably what cases the Tsan failure (yay).

Is there a reason we can't just schedule a Matter timer to call `NextTest()`?  That seems like it would closely mirror how `NextTest()` would normally get called mid-test: off a callback running on the Matter event loop.

Reply:
I used a Matter timer. Thanks for the idea.